include/psen_scan_v2/scanner_state_machine_def.h

```diff
@@ -146,15 +146,29 @@ inline void ScannerProtocolDef::handleMonitoringFrame(const scanner_events::RawM
 {
   PSENSCAN_DEBUG("StateMachine", "Action: handleMonitoringFrame");
   monitoring_frame_watchdog_->reset();
-  const monitoring_frame::Message frame{ monitoring_frame::deserialize(event.data_, event.num_bytes_) };
-  if (!frame.diagnosticMessages().empty())
+
+  try
   {
-    PSENSCAN_WARN_THROTTLE(
-        1 /* sec */, "StateMachine", "The scanner reports an error: {}", formatRange(frame.diagnosticMessages()));
+    const monitoring_frame::Message frame{ monitoring_frame::deserialize(event.data_, event.num_bytes_) };
+    if (!frame.diagnosticMessages().empty())
+    {
+      PSENSCAN_WARN_THROTTLE(
+          1 /* sec */, "StateMachine", "The scanner reports an error: {}", formatRange(frame.diagnosticMessages()));
+    }
+
+    printUserMsgFor(complete_scan_validator_.validate(frame, DEFAULT_NUM_MSG_PER_ROUND));
+    args_->inform_user_about_laser_scan_cb(toLaserScan(frame));
   }
-
-  printUserMsgFor(complete_scan_validator_.validate(frame, DEFAULT_NUM_MSG_PER_ROUND));
-  args_->inform_user_about_laser_scan_cb(toLaserScan(frame));
+  // LCOV_EXCL_START
+  catch (const scanner_reply::CRCMismatch& e)
+  {
+    PSENSCAN_ERROR("StateMachine", e.what());
+  }
+  catch (const monitoring_frame::ScanCounterMissing& e)
+  {
+    PSENSCAN_ERROR("StateMachine", e.what());
+  }
```

Comment:
The CRCMismatch exception is not thrown here, but below from the scanner reply deserialization. E.g. here:
```
  const scanner_reply::Message msg{ scanner_reply::deserialize(reply_event.data_) };
  check_for_internal_errors(msg);
  return msg.type() == scanner_reply::Message::Type::start;
```

Has anybody an idea what happens if the exception is thrown? I guess we have to actively transition into an error state, else we leave the state machine in an undetermined state.